hugr-core/src/extension/op_def.rs

```diff
@@ -147,21 +139,22 @@ impl CustomValidator {
     }
 }
 
-/// The two ways in which an OpDef may compute the Signature of each operation node.
-#[derive(serde::Deserialize, serde::Serialize)]
+/// The ways in which an OpDef may compute the Signature of each operation node.
 pub enum SignatureFunc {
-    // Note: except for serialization, we could have type schemes just implement the same
-    // CustomSignatureFunc trait too, and replace this enum with Box<dyn CustomSignatureFunc>.
-    // However instead we treat all CustomFunc's as non-serializable.
-    /// A PolyFuncType (polymorphic function type), with optional custom
-    /// validation for provided type arguments,
-    #[serde(rename = "signature")]
-    PolyFuncType(CustomValidator),
-    #[serde(skip)]
+    /// An explicit polymorphic function type.
+    PolyFuncType(PolyFuncTypeRV),
+    /// A polymorphic function type with a custom binary for validating type arguments.
+    CustomValidator(CustomValidator),
+    /// Serialized declaration specified a custom validate binary but it was not provided.
+    MissingValidateFunc(PolyFuncTypeRV),
     /// A custom binary which computes a polymorphic function type given values
     /// for its static type parameters.
     CustomFunc(Box<dyn CustomSignatureFunc>),
+    /// Serialized declaration specified a custom compute binary but it was not provided.
+    MissingComputeFunc,
 }
+
+#[derive(PartialEq, Eq, Debug)]
```

Comment:
Methinks we should not need `NoValidate`. One should use `Signature::PolyFuncType` instead.

Reply:
yes, moved it just to the test that used it